public/Aframe/samples/text-input-puzzles/components.js

```diff
@@ -0,0 +1,84 @@
+var input = "";
+var key = "time";
+
+var prompt = `<a-box src="#riddle" static-body position="0 12 -20" width="10" height="10" depth="1"></a-box>`;
+
+function setPrompt(e) {}
+
+function updateInput(e) {
+    var code = parseInt(e.detail.code);
+    console.log(code);
+    document.querySelector("#input").setAttribute("color", "black");
+    switch (code) {
```

Comment:
Without really going into the logic here, what's the difference between the codes (are they just different characters) and why do some use a return and some use a break? Also, are these the only codes that we need to specifically handle in a special way?

Reply:
Yep, so for example 8 corresponds to the 'delete' key and 13 corresponds to 'return'/'enter'. The delete key needed a special case so that the input text would slice off the last character, and the return needed a special case for submitting and checking. AFAIK there are no other keys that need special treatment (other than Esc, and potentially mouse click, which is WIP). However, I haven't fully stress-tested yet.

For return vs break, that's mostly because after the switch statement I have a couple of lines to update the text to whatever character has been added and then add a trailing underscore '_' which was meant to represent a typing "cursor". Cases where I use return were for when the user got the answer right or wrong - in that case, I didn't want to update the text with the user's input, but rather a red X or congratulations and return that. If this is bad style, I can fix things to either do all return's or all break's.

Reply:
No, this is very valid. Thanks for the detailed explanation, and great idea for the _ being a current character symbol!